Docs: Add multi-dimension partitioning doc; refactor native batch and separate into smaller topics. (#11983)
apache/druid: docs/ingestion/native-batch.md

> in `partial dimension distribution` and `partial segment generation` phases,
> the task may fail if the input changes in between the two passes.

#### Multi-dimension range partitioning
> Multiple dimension (multi-dimension) range partitioning is an experimental feature. Multi-dimension range partitioning is currently not supported in the sequential mode of the Parallel task.

When you use multi-dimension partitioning for your data, Druid is able to distribute segment sizes more evenly than with single dimension partitioning.

> For segment pruning to be effective and translate into better query performance, you must the first of your `partitionDimensions` at query time. For example, given the following `partitionDimensions`:

Reviewer: I think a word(s) is missing here, but not sure what it should be:  "...you must __ the first..."

Returns the task attempt history of the worker task spec of the given id, or HTTP 404 Not Found error if the supervisor task is running in the sequential mode.

### Segment pushing modes
While ingesting data using the parallel task indexing, Druid creates segments from the input data and pushes them. For segment pushing,
the parallel task index supports the following segment pushing modes based upon your type of [rollup](./rollup.md):

- Bulk pushing mode: Used for perfect rollup. Druid pushes every segment at the very end of the index task. Until then, Druid stores created segments in memory and local storage of the service running the index task. This mode can cause problems if you have limited storage capacity, and is not recommended to use in production.

> To enable bulk pushing mode, set `forceGuaranteedRollup` in your TuningConfig. You can not use bulk pushing with `appendToExisting` in your IOConfig.

Reviewer: ```suggestion
To enable bulk pushing mode, set `forceGuaranteedRollup` in your TuningConfig. You cannot use bulk pushing with `appendToExisting` in your IOConfig.
```
